File: app/src/main/java/com/xpf/aop_example/click/SingleClick.java
package com.xpf.aop_example.click;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by x-sir on 2019-05-25 :)
 * Function:
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface SingleClick {
    /**
     * 点击间隔时间
     */
    long value() default 1000;
}

File: app/src/main/java/com/xpf/aop_example/SingleClick.kt
package com.xpf.aop_example

/**
 * Created by x-sir on 2019-05-24 :)
 * Function:自定义单击注解(可传递点击时长，默认为1秒)
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class SingleClick(val value: Long = 1000)
